Source/Libraries/openXDA.Nodes/Types/Analysis/AnalysisNode.cs

```diff
@@ -306,19 +312,51 @@ private void Process(MeterDataSet meterDataSet)
                     }
 
                     Log.Debug($"Finished executing data operation '{dataOperationDefinition.UnqualifiedTypeName}'.");
+                    if (index == 0) status = FileGroupProcessingStatus.Success;
+                    else if (status == FileGroupProcessingStatus.Failed) status = FileGroupProcessingStatus.PartialSuccess;
```

Comment:
This logic seems off. 
If it has 3 `dataOperationDefinitions` (and all suceed) it will go:
from `Failure` to `Success` in `index = 0` 
not change in `index = 1` and `index = 2`

But if the last one fails it will go 
from `Failure` to `Success` in `index = 0` 
not change in `index = 1` and 
from `Success` to `Failure` in `index = 2` which is not correct.

Reply:
With the current code, on failure it should either
Be in `Failure` or `Partial Success`, in either case things shouldn't change.
Be in `Success` and go to `Partial Success`, as is the case on line 323. 

In the case you highlighted, the code I believe is doing the following.
Go from `Failure` to `Success` in `index = 0` (line 315)
Not change in `index = 1` (line  315-316, neither condition should be met)
Go from `Success` to `Partial Success` in `index = 2` (line 323)